src/librados/librados_c.cc

```diff
@@ -2898,6 +2898,81 @@ extern "C" int _rados_notify2(rados_ioctx_t io, const char *o,
 }
 LIBRADOS_C_API_BASE_DEFAULT(rados_notify2);
 
+extern "C" int _rados_decode_notify_msg(char *reply_buffer, size_t reply_buffer_len,
+                                        struct notify_ack_t **acks, size_t *nr_acks,
+                                        struct notify_timeout_t **timeouts, size_t *nr_timeouts) {
+  if (!reply_buffer || !reply_buffer_len) {
+    return -EINVAL;
+  }
+
+  bufferlist bl;
+  try{
+    bufferptr p = buffer::create(reply_buffer_len);
+    memcpy(p.c_str(), reply_buffer, reply_buffer_len);
+    bl.push_back(p);
+  } catch (const ceph::buffer::error &e) {
+    return -EINVAL;
+  }
+
+  map<pair<uint64_t,uint64_t>,bufferlist> acked;
+  set<pair<uint64_t,uint64_t>> missed;
+  auto iter = bl.cbegin();
+  decode(acked, iter);
+  decode(missed, iter);
+
+  *nr_acks = acked.size();
+  if (acked.size()) {
+    *acks = (struct notify_ack_t *)malloc(sizeof(struct notify_ack_t) * acked.size());
```

Comment:
how about resetting `*acks` if `acked.empty()`?

Reply:
Probably -- since we touch `nr_acks` anyway. So just `*acks = nullptr` and allocate it when there is something to copy.

Reply:
The python binding set *acks and *timeout to NULL before calling this. So, is there a need to explicitly reset it?